### ISSUE TITLE
unvisible text on VMware Workstation

### ISSUE THREAD

**Issue by ceyhunnabiyev** [tista500/Adapta#246]
When I run vmware from terminal, it gives following errors:

/usr/share/themes/Adapta-Nokto-Eta/gtk-2.0/common-dark.rc:1136: error: unexpected identifier `direction', expected character `}'
/usr/share/themes/Adapta-Nokto-Eta/gtk-2.0/misc-dark.rc:36: error: unexpected identifier `direction', expected character `}'
/usr/share/themes/Adapta-Nokto-Eta/gtk-2.0/3rd-party-dark.rc:341: error: invalid string constant "standalone_entry", expected valid string constant
/usr/share/themes/Adapta-Nokto-Eta/gtk-2.0/common-dark.rc:1136: error: unexpected identifier `direction', expected character `}'
/usr/share/themes/Adapta-Nokto-Eta/gtk-2.0/misc-dark.rc:36: error: unexpected identifier `direction', expected character `}'
/usr/share/themes/Adapta-Nokto-Eta/gtk-2.0/3rd-party-dark.rc:341: error: invalid string constant "standalone_entry", expected valid string constant
/usr/share/themes/Adapta-Nokto-Eta/gtk-2.0/common-dark.rc:1136: error: unexpected identifier `direction', expected character `}'
/usr/share/themes/Adapta-Nokto-Eta/gtk-2.0/misc-dark.rc:36: error: unexpected identifier `direction', expected character `}'
/usr/share/themes/Adapta-Nokto-Eta/gtk-2.0/3rd-party-dark.rc:341: error: invalid string constant "standalone_entry", expected valid string constant

![screenshot from 2016-10-14 01-26-04](https://cloud.githubusercontent.com/assets/5506156/19367798/c25ebe62-91ad-11e6-992c-c69006dff431.png)

**Comment by tista500** [comment]
Is it REALLY theme issue?
`direction` property in Gtk+ 2.x is standard feature, and I don't see such things in any native Gtk+ 2.x applications.

Regards.

**Comment by ceyhunnabiyev** [comment]
unvisible text occured due to these errors or other reason??

**Comment by ceyhunnabiyev** [comment]
in Vlc media player also occured unvisible text and too dark blue link color

![screenshot from 2016-10-14 01-51-39](https://cloud.githubusercontent.com/assets/5506156/19368406/ef7edffa-91b0-11e6-9029-
3c39cfe3e5cc.png)

**Comment by tista500** [comment]
Hey, please stop putting multiple issues in a single report and why you ignored issue-templates?

Invalid.

**Comment by ceyhunnabiyev** [comment]
Sorry about that..

**Comment by tista500** [comment]
VMware draws its own images inside their tabs as "background", it causes foreground of text labels are dimmed and it can't be fixed from Gtk+ theme. It's the part of the issue of VMware Gtk+ backend.

Vlc is Qt application, to show it, we need Qt-Gtk conversion backend (installed by default in recent modern destribution). And that backend can't recognize the styling: "If Buttons are selected, foreground colour is pulled from `fg[SELECTED]` and `text[SELECTED]` property from Gtk+ theming". This is the limitation of current Qt-Gtk backend. Then `@link_color`, I don't have any clue...

So finally almost "un-fixable".

Regards.

**Comment by ceyhunnabiyev** [comment]
Thanks for attention... I don't know  why  famous applications like these use gtk2 backend, not gtk3..

**Comment by GhettoGirl** [comment]
Not sure whats going on here, but for me VMware itself doesn't allow theming and discards every modification at startup and shows its **own** theme, regardless of what theme is currenty configured.

Did you deleted VMware internal Gtk libs or patched the startup/init script in some way?

**Comment by ceyhunnabiyev** [comment]
No, where did you delete from?? Do you see any errors when running vmware from terminal, @GhettoGirl  ??

**Comment by GhettoGirl** [comment]
@ceyhunnabiyev 

This is the terminal output:
```
# >  vmware
Gtk-Message: Failed to load module "canberra-gtk-module": libcanberra-gtk-module.so: cannot open shared object file: No such file or directory
Gtk-Message: Failed to load module "canberra-gtk-module": libcanberra-gtk-module.so: cannot open shared object file: No such file or directory
```

And this is the memory map, as you can see here VMware comes with its own GTK libraries. As there is no theming I guess they patched it and removed everything related to themes.
```
18120 KB	[heap]
4064 KB	/usr/lib/vmware/lib/libgvmomi.so/libgvmomi.so
3684 KB	/usr/lib/vmware/lib/libvmware.so/libvmware.so
2840 KB	/usr/lib/vmware/lib/libvmwareui.so/libvmwareui.so
640 KB	/usr/lib/vmware/lib/libfreetype.so.6/libfreetype.so.6
620 KB	/usr/lib/vmware/lib/libgtkmm-2.4.so.1/libgtkmm-2.4.so.1
612 KB	/usr/lib/vmware/lib/libvmwarebase.so/libvmwarebase.so
312 KB	/usr/lib/vmware/lib/libcurl.so.4/libcurl.so.4
312 KB	/usr/lib/vmware/lib/libgtk-x11-2.0.so.0/libgtk-x11-2.0.so.0
284 KB	/usr/lib/vmware/lib/libxml2.so.2/libxml2.so.2
280 KB	/usr/lib/vmware/lib/libcrypto.so.1.0.1/libcrypto.so.1.0.1
148 KB	/usr/lib/vmware/lib/libglib-2.0.so.0/libglib-2.0.so.0
148 KB	[stack]
136 KB	/usr/lib/vmware/lib/libpixman-1.so.0/libpixman-1.so.0
112 KB	/usr/lib/vmware/lib/libstdc++.so.6/libstdc++.so.6
112 KB	/usr/lib/vmware/lib/libcairo.so.2/libcairo.so.2
96 KB	/usr/lib/vmware/bin/appLoader
76 KB	/usr/lib/vmware/lib/libcds.so/libcds.so
68 KB	/usr/lib/vmware/lib/libview.so.3/libview.so.3
68 KB	/usr/lib/vmware/lib/libpangoft2-1.0.so.0/libpangoft2-1.0.so.0
56 KB	/usr/lib/vmware/lib/libgiomm-2.4.so.1/libgiomm-2.4.so.1
44 KB	/usr/lib/vmware/lib/libgcr.so.0/libgcr.so.0
40 KB	/usr/lib/vmware/lib/libssl.so.1.0.1/libssl.so.1.0.1
36 KB	/usr/lib/libasound.so.2.0.0
36 KB	/usr/lib/vmware/lib/libgdkmm-2.4.so.1/libgdkmm-2.4.so.1
36 KB	/usr/lib/vmware/lib/libpango-1.0.so.0/libpango-1.0.so.0
32 KB	/usr/lib/vmware/lib/libglibmm-2.4.so.1/libglibmm-2.4.so.1
28 KB	/usr/lib/vmware/lib/libpangomm-1.4.so.1/libpangomm-1.4.so.1
28 KB	/usr/lib/vmware/lib/libgdk-x11-2.0.so.0/libgdk-x11-2.0.so.0
28 KB	/usr/lib/vmware/lib/libgobject-2.0.so.0/libgobject-2.0.so.0
24 KB	/usr/lib/vmware/lib/libatkmm-1.6.so.1/libatkmm-1.6.so.1
24 KB	/usr/lib/libX11.so.6.3.0
24 KB	/usr/lib/vmware/lib/libexpat.so.0/libexpat.so.0
24 KB	/usr/lib/vmware/lib/libgio-2.0.so.0/libgio-2.0.so.0
24 KB	/usr/lib/libc-2.24.so
16 KB	/usr/lib/vmware/icu/icudt44l.dat
16 KB	/usr/lib/libgcrypt.so.11.8.5
16 KB	/usr/lib/vmware/lib/libcairomm-1.0.so.1/libcairomm-1.0.so.1
16 KB	/usr/lib/vmware/lib/librsvg-2.so.2/librsvg-2.so.2
16 KB	/usr/lib/libfontconfig.so.1.9.2
12 KB	/usr/lib/vmware/libconf/lib/pango/1.6.0/modules/pango-basic-fc.so
12 KB	/usr/lib/vmware/lib/libatk-1.0.so.0/libatk-1.0.so.0
12 KB	/usr/lib/libexpat.so.1.6.2
8 KB	/usr/lib/gconv/UTF-16.so
8 KB	/usr/lib/libgpg-error.so.0.19.1
8 KB	/usr/lib/vmware/lib/libgck.so.0/libgck.so.0
8 KB	/usr/lib/vmware/lib/libbasichttp.so/libbasichttp.so
8 KB	/usr/lib/libXrandr.so.2.2.0
8 KB	/usr/lib/libXext.so.6.4.0
8 KB	/usr/lib/vmware/lib/libgcc_s.so.1/libgcc_s.so.1
8 KB	/usr/lib/libXdmcp.so.6.0.0
8 KB	/usr/lib/libXau.so.6.0.0
8 KB	/usr/lib/libxcb.so.1.1.0
8 KB	/usr/lib/vmware/lib/libpng12.so.0/libpng12.so.0
8 KB	/usr/lib/libm-2.24.so
8 KB	/usr/lib/libresolv-2.24.so
8 KB	/usr/lib/librt-2.24.so
8 KB	/usr/lib/libpcre.so.1.2.7
8 KB	/usr/lib/libglib-2.0.so.0.5000.1
8 KB	/usr/lib/libnss_files-2.24.so
8 KB	/usr/lib/libgcc_s.so.1
8 KB	/usr/lib/libpthread-2.24.so
8 KB	/usr/lib/libdl-2.24.so
8 KB	/usr/lib/ld-2.24.so
4 KB	/usr/lib/vmware/libconf/lib/gtk-2.0/2.10.0/loaders/svg_loader.so
4 KB	/usr/lib/vmware/libconf/lib/gtk-2.0/2.10.0/loaders/libpixbufloader-png.so
4 KB	/usr/lib/vmware/libconf/lib/gtk-2.0/2.10.0/engines/libclearlooks.so
4 KB	/usr/lib/vmware/lib/libXdamage.so.1/libXdamage.so.1
4 KB	/usr/lib/vmware/lib/libXcomposite.so.1/libXcomposite.so.1
4 KB	/usr/lib/vmware/lib/libXcursor.so.1/libXcursor.so.1
4 KB	/usr/lib/vmware/lib/libXfixes.so.3/libXfixes.so.3
4 KB	/usr/lib/vmware/lib/libXinerama.so.1/libXinerama.so.1
4 KB	/usr/lib/vmware/lib/libsigc-2.0.so.0/libsigc-2.0.so.0
4 KB	/usr/lib/vmware/lib/libpangocairo-1.0.so.0/libpangocairo-1.0.so.0
4 KB	/usr/lib/vmware/lib/libXrender.so.1/libXrender.so.1
4 KB	/usr/lib/vmware/lib/libgdk_pixbuf-2.0.so.0/libgdk_pixbuf-2.0.so.0
4 KB	/usr/lib/vmware/lib/libz.so.1/libz.so.1
4 KB	/usr/lib/vmware/lib/libgmodule-2.0.so.0/libgmodule-2.0.so.0
4 KB	/usr/lib/vmware/lib/libgthread-2.0.so.0/libgthread-2.0.so.0
```

When you remove this libraries it will fallback to the system installed GTK which has support for theming and therefore Adapta is loaded and applied to VMware.

Where and how did you installed VMware? Does all this libraries exist in `/usr/lib/vmware`? If not try reinstalling from the official bundle - this should solve the problem. You can also try to unset `LD_LIBRARY_PATH` before launching vmware to see what happens then.

In general there is no theme support in VMware when you installed it correctly. And I recommend to follow this suggestion from the developers.

**Comment by TheAifam5** [comment]
``Gtk-Message: Failed to load module "canberra-gtk-module":``
On Arch just set ``GTK_PATH=:/usr/lib/gtk-2.0`` and will repair that problem.

**Comment by hogar1977** [comment]
@TheAifam5 where in that shell script did you change this exactly...

I will not pretend that I understand everything that is in that script, but basically the gist of it looks like its using the shipped libraries for a reason - stability, etc...
That being said, the the broken Adapta theme really rubs me the wrong way and would appreciate any pointers on how to solve it. I opened the issue that tista500 referenced above and based on this discussion I also tried to run vmplayer from terminal and I get roughly the same message as @ceyhunnabiyev.

And yes, it seems that VMware are using their own set of libraries which are causing this strange behavior... What would happen if I just removed those libraries (moved them somewhere else) and than try to run vmplayer? If it doesn't work I can just put the libraries back, right? No harm done?
